docs/latest/conf.py

```diff
@@ -10,6 +9,9 @@
     'myst_parser',
 ]
 
+autosectionlabel_prefix_document = True
```

Comment:
I was able to fix the references to sections by removing the `sphinx.ext.autosectionlabel` extension, instead of setting `autosectionlabel_prefix_document`, following instructions I found in https://stackoverflow.com/a/71660856. I'm not sure this is necessarily a better option, but I thought it was worth mentioning it.

Reply:
it appears `sphinx.ext.autosectionlabel ` is relevant only when using `:ref:` which when grepping for, couldn't find any usages. But we do have rst docs, mostly as directory pages, so not sure if its worth keeping around in case it does come up and we want to link back to an section within them.